src/models/calendar.js

```diff
@@ -75,39 +74,92 @@ export const getDefaultCalendarObject = (props = {}) => Object.assign({}, {
  * Map a dav collection to our calendar object model
  *
  * @param {Object} calendar The calendar object from the cdav library
+ * @param {Object=} currentUserPrincipal The principal model of the current user principal
  * @returns {Object}
  */
-export function mapDavCollectionToCalendar(calendar) {
-	let color = calendar.color || uidToHexColor(calendar.displayname)
-	if (color.length === 9) {
-		// Make sure it's #RRGGBB, not #RRGGBBAA
-		color = color.substr(0, 7)
+export function mapDavCollectionToCalendar(calendar, currentUserPrincipal) {
+	const id = btoa(calendar.url)
+	const displayName = calendar.displayname
+
+	// calendar.color can be set to anything on the server,
+	// so make sure it's something that remotely looks like a color
+	let color
```

Comment:
This whole part might as well be moved to `utils/color.js` in a `detectColor` function.

Reply:
Done 👍